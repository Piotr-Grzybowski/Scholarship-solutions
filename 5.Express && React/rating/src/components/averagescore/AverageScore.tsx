import { ratings, rating } from "../types";
import Score from "../score/Score";

function AverageScore({ ratings }: ratings) {
  return (
    <div>
      <span>Average score: </span>
      <Score score={getAverageScore(ratings)}></Score>
      <hr></hr>
    </div>
  );
}

export default AverageScore;

function getAverageScore(array: rating[]) {
  return Math.floor(
    array.reduce((acc, element) => {
      return (acc += element.score);
    }, 0) / array.length
  );
}
