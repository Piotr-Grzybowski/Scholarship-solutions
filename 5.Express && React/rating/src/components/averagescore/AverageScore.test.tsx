import { render, screen } from "@testing-library/react";
import AverageScore from "./AverageScore";
import renderer from "react-test-renderer";

describe("Testing AverageScore", () => {
  const data = [
    {
      recordId: "1",
      name: "",
      score: 3,
      content: "",
    },
    {
      recordId: "2",
      name: "",
      score: 5,
      content: "",
    },
    {
      recordId: "3",
      name: "",
      score: 0,
      content: "",
    },
  ];

  test("when given data, average score should render proper amount of stars", () => {
    render(<AverageScore ratings={data}></AverageScore>);
    const greyStars = screen.getAllByTestId("gray-star");
    const yellowStars = screen.queryAllByTestId("yellow-star");
    expect(greyStars.length).toBe(3);
    expect(yellowStars.length).toBe(2);
  });
  test("when given data average score should create proper snapshot", () => {
    const component = renderer
      .create(<AverageScore ratings={data}></AverageScore>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
