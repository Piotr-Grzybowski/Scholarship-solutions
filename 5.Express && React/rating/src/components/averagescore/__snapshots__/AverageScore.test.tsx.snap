// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AverageScore when given data average score should create proper snapshot 1`] = `
<div
  data-testid="average-score"
>
  <span>
    Average score: 
  </span>
  <div
    data-testid="score"
  >
    <svg
      data-testid="yellow-star"
    >
      yellowStar.svg
    </svg>
    <svg
      data-testid="yellow-star"
    >
      yellowStar.svg
    </svg>
    <svg
      data-testid="gray-star"
    >
      grayStar.svg
    </svg>
    <svg
      data-testid="gray-star"
    >
      grayStar.svg
    </svg>
    <svg
      data-testid="gray-star"
    >
      grayStar.svg
    </svg>
  </div>
  <hr />
</div>
`;
